src/Appwrite/Extend/Exception.php

```diff
@@ -0,0 +1,186 @@
+     * <ENTITY>_<ERROR_TYPE>
+     * 
+     * Appwrite has the follwing entities:
+     * - Users
+     * - Projects
+     * - Sessions
+     * - Teams
+     * - Memberships
+     * - Files
+     * - Functions
+     * - Deployments
+     * - Executions
+     */
+
+    /** Users */
+    const USER_COUNT_EXCEEDED               = 'user_count_exceeded';
+    const USER_ALREADY_EXISTS               = 'user_already_exists';
+    const USER_BLOCKED                      = 'user_blocked';
+    const USER_CREATION_FAILED              = 'user_creation_failed';
+    const USER_INVALID_TOKEN                = 'user_invalid_token';
+    const USER_NOT_FOUND                    = 'user_not_found';
+    const USER_INVALID_CREDENTIALS          = 'user_invalid_credentials';
+    const USER_EMAIL_ALREADY_EXISTS         = 'user_email_already_exists';
+    const USER_PASSWORD_MISMATCH            = 'user_password_mismatch';
+    const USER_AUTH_METHOD_UNSUPPORTED      = 'user_auth_method_unsupported';
+    const USER_PASSWORD_RESET_REQUIRED      = 'user_password_reset_required';
+    const USER_EMAIL_NOT_WHITELISTED        = 'user_email_not_whitelisted';
+    const USER_IP_NOT_WHITELISTED           = 'user_ip_not_whitelisted';
+    const USER_SESSION_ALREADY_EXISTS       = 'user_session_already_exists';
+    const USER_SESSION_NOT_FOUND            = 'user_session_not_found';
+    const USER_UNAUTHORIZED                 = 'user_unauthorized';
+    const USER_ANONYMOUS_CONSOLE_PROHIBITED = 'user_anonymous_console_prohibited';
+
+    /** OAuth **/
+    const OAUTH_PROVIDER_DISABLED          = 'oauth_provider_disabled';
+    const OAUTH_PROVIDER_UNSUPPORTED       = 'oauth_provider_unsupported';
+    const OAUTH_INVALID_LOGIN_STATE_PARAMS = 'oauth_invalid_login_state_params';
+    const OAUTH_INVALID_SUCCESS_URL        = 'oauth_invalid_success_url';
+    const OAUTH_INVALID_FAILURE_URL        = 'oauth_invalid_failure_url';
+    const OAUTH_ACCESS_TOKEN_FAILED        = 'oauth_access_token_failed';
+    const OAUTH_MISSING_USER_ID            = 'oauth_missing_user_id';
+
+    /** Teams */
+    const TEAM_NOT_FOUND                 = 'team_not_found';
+    const TEAM_DELETION_FAILED           = 'team_deletion_failed';
+    const TEAM_INVITE_ALREADY_EXISTS     = 'team_invite_already_exists';
+    const TEAM_INVITE_NOT_FOUND          = 'team_invite_not_found';
+    const TEAM_INVALID_SECRET            = 'team_invalid_secret';
+    const TEAM_MEMBERSHIP_MISMATCH       = 'team_membership_mismatch';
+    const TEAM_INVITE_MISMATCH           = 'team_invite_mismatch'; 
+
+    /** Membership */
+    const MEMBERSHIP_NOT_FOUND       = 'membership_not_found';
+    const MEMBERSHIP_DELETION_FAILED = 'membership_deletion_failed';
+
+    /** Avatars */
+    const AVATAR_SET_NOT_FOUND             = 'avatar_set_not_found';
+    const AVATAR_NOT_FOUND                 = 'avatar_not_found';
+    const IMAGIC_EXTENSION_MISSING         = 'imagic_extension_missing';
```

Comment:
I think for server errors (500) we shouldn't share any error type or give one generic type.

Reply:
done 👍